cmd/gopogh-server/flake_chart.js

```diff
@@ -0,0 +1,798 @@
+                  minValue: 0
+              },
+              1: {
+                  title: "Duration (seconds)",
+                  minValue: 0
+              },
+          },
+          tooltip: {
+              trigger: "selection",
+              isHtml: true
+          }
+      };
+      const envDurationContainer = document.createElement("div");
+      envDurationContainer.style.width = "100vw";
+      envDurationContainer.style.height = "100vh";
+      chartsContainer.appendChild(envDurationContainer);
+      const durChart = new google.visualization.LineChart(envDurationContainer);
+      durChart.draw(durationChart, durOptions);
+  }
+
+  chartsContainer.appendChild(createRecentFlakePercentageTable(data.recentFlakePercentTable, query))
+}
+
+function createTopnDropdown(currentTopn) {
+  const dropdownContainer = document.createElement("div");
+  dropdownContainer.style.margin = "1rem";
+
+  const dropdownLabel = document.createElement("label");
+  dropdownLabel.innerText = "Select topn value: ";
+
+  const dropdown = document.createElement("select");
+  dropdown.id = "topnDropdown";
+
+  const values = [3, 5, 10, 15];
+  values.forEach(value => {
+      const option = document.createElement("option");
+      option.value = value;
+      option.text = value;
+      if (value.toString() === currentTopn) {
+          option.selected = true;
+      }
+      dropdown.appendChild(option);
+  });
+
+  dropdown.addEventListener("change", () => {
+      const selectedValue = dropdown.value;
+      const currentURL = new URL(window.location.href);
+      currentURL.searchParams.set("tests_in_top", selectedValue);
+      window.location.href = currentURL.href;
+  });
+
+  dropdownContainer.appendChild(dropdownLabel);
+  dropdownContainer.appendChild(dropdown);
+
+  document.getElementById('dropdown_container').appendChild(dropdownContainer)
+}
+
+
+async function init() {
+  const query = parseUrlQuery(window.location.search);
+  const desiredTest = query.test,
+      desiredEnvironment = query.env,
+      desiredPeriod = query.period || "",
+      desiredTestNumber = query.tests_in_top || "";
+  const currentTopn = query.tests_in_top || "10"; // Default to 10 (for top 10 tests)
+
+  google.charts.load('current', {
+      'packages': ['corechart']
+  });
+  try {
+      // Wait for Google Charts to load
+      await new Promise(resolve => google.charts.setOnLoadCallback(resolve));
+
+      let url;
+      if (desiredEnvironment === undefined) {
+          // URL for displaySummaryChart
+          url = 'http://localhost:8080/summary'
+      } else if (desiredTest === undefined) {
+          // URL for displayEnvironmentChart
+          url = 'http://localhost:8080/env' + '?env=' + desiredEnvironment + '&tests_in_top=' + desiredTestNumber;
+      } else {
+          // URL for displayTestAndEnvironmentChart
+          url = 'http://localhost:8080/test' + '?env=' + desiredEnvironment + '&test=' + desiredTest;
+      }
```

Comment:
There localhost URLs aren't going to work once this file is hosted in GCS and someone opens it, it will need to hit the cloudrun instance

Reply:
maybe you could use var `currentURL = window.location.href;`

Reply:
maybe you could use var `currentURL = window.location.href;`

Reply:
Noted: Gopogh is independent of MK so we just have the localhost as a placeholder for whoever may be using it